src/main/java/slimeknights/tconstruct/tools/harvest/KamaTool.java

```diff
@@ -0,0 +1,253 @@
+      Material.PLANTS,
+      Material.GOURD,
+      Material.CACTUS,
+      Material.BAMBOO);
+
+  public KamaTool(Properties properties, ToolDefinition toolDefinition) {
+    super(properties, toolDefinition);
+  }
+
+  @Override
+  public boolean isEffective(BlockState state) {
+    return effective_materials.contains(state.getMaterial());
+  }
+
+  @Override
+  public ActionResultType onItemUse(ItemUseContext context) {
+    PlayerEntity playerentity = context.getPlayer();
+    ItemStack itemStack = playerentity.getHeldItem(context.getHand());
+
+    if (ToolData.isBroken(itemStack)) {
+      return ActionResultType.FAIL;
+    }
+
+    ActionResultType resultType = Items.DIAMOND_HOE.onItemUse(context);
+
+    if (resultType == ActionResultType.SUCCESS) {
+      //TODO event
+    }
+
+    return resultType;
+  }
+
+  @Override
+  public boolean itemInteractionForEntity(ItemStack stack, PlayerEntity playerIn, LivingEntity target, Hand hand) {
+    // only run AOE on shearable entities
+    if (target instanceof IShearable) {
+      int fortune = EnchantmentHelper.getEnchantmentLevel(Enchantments.FORTUNE, stack);
+
+      if (this.shearEntity(stack, playerIn.getEntityWorld(), playerIn, target, fortune)) {
+        this.swingTool(playerIn, hand);
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  protected void swingTool(PlayerEntity player, Hand hand) {
+    player.swingArm(hand);
+    player.spawnSweepParticles();
+  }
+
+  public boolean shearEntity(ItemStack itemStack, World world, PlayerEntity playerEntity, Entity entity, int fortune) {
+    if (!(entity instanceof IShearable)) {
+      return false;
+    }
+
+    IShearable target = (IShearable) entity;
+
+    if (target.isShearable(itemStack, world, entity.getPosition())) {
+      if (!world.isRemote) {
+        List<ItemStack> drops = target.onSheared(itemStack, world, entity.getPosition(), fortune);
+        Random rand = world.rand;
+
+        drops.forEach(d -> {
+          ItemEntity ent = entity.entityDropItem(d, 1.0F);
+
+          if (ent != null) {
+            ent.setMotion(ent.getMotion().add((rand.nextFloat() - rand.nextFloat()) * 0.1F, rand.nextFloat() * 0.05F, (rand.nextFloat() - rand.nextFloat()) * 0.1F));
+          }
+        });
+      }
+
+      ToolInteractionUtil.damageTool(itemStack, 1, playerEntity);
+
+      return true;
+    }
+
+    return false;
+  }
+
+  public boolean harvestCrop(ItemStack stack, World world, PlayerEntity player, BlockPos pos, int fortune) {
+    BlockState state = world.getBlockState(pos);
+
+    boolean canHarvest = this.canHarvestCrop(state);
+
+    // do not harvest bottom row reeds
+    if (state.getBlock() instanceof SugarCaneBlock && !(world.getBlockState(pos.down()).getBlock() instanceof SugarCaneBlock)) {
+      canHarvest = false;
+    }
+
+    if (!canHarvest) {
+      return false;
+    }
+
+    // can be harvested, always just return true clientside for the animation stuff
+    if (!world.isRemote) {
```

Comment:
I would prefer `world instanceof ServerWorld` for a server world check if you cast it down the line, mods can make a custom world implementation that is not `ServerWorld` while being non-remote.